### PR TITLE
Issues/479

### LINKED ISSUE #479: Issues displaying Umlaute in welcome-screen
image

IT's supposed to read "**dämmerung**"

### PR DESCRIPTION
**Pull request by reusr1**
fixes umlaut issue in create tenant call. also contains PR #656

### CODE REVIEW

**Comment by flx-sta** [comment]
@reusr1 is this supposed to be a draft due to waiting for #656 ?

File: platform/commons/src/main/java/com/peregrine/commons/servlets/AbstractBaseServlet.java [hunk 1]
```diff
@@ -147,11 +148,19 @@ private void doRequest(SlingHttpServletRequest request, SlingHttpServletResponse
...
+
         private final SlingHttpServletRequest request;
         private final SlingHttpServletResponse response;
         private final Map<String, String> parameters;
 
         public Request(SlingHttpServletRequest request, SlingHttpServletResponse response) {
+            try {
+                request.setCharacterEncoding("UTF-8");
+            } catch( UnsupportedEncodingException uee ) {
+                logger.info("not able to set character decoding to UTF-8");
```
Comment (flx-sta) [inline platform/commons/src/main/java/com/peregrine/commons/servlets/AbstractBaseServlet.java]: should this really be only `logger.info` ? I'd rather say this is a warning

**Comment by reusr1** [comment]
@Felix-Puetz yes, just waiting on #656 to be merged

**Comment by flx-sta** [comment]
@reusr1 #659 has been merged

File: platform/commons/src/main/java/com/peregrine/commons/servlets/AbstractBaseServlet.java [hunk 2]
```diff
@@ -147,11 +148,19 @@ private void doRequest(SlingHttpServletRequest request, SlingHttpServletResponse
...
+
+        protected final Logger logger = LoggerFactory.getLogger(getClass());
+
         private final SlingHttpServletRequest request;
         private final SlingHttpServletResponse response;
         private final Map<String, String> parameters;
 
         public Request(SlingHttpServletRequest request, SlingHttpServletResponse response) {
+            try {
+                request.setCharacterEncoding("UTF-8");
```
Comment (reggie7) [inline platform/commons/src/main/java/com/peregrine/commons/servlets/AbstractBaseServlet.java]: We could use `com.peregrine.commons.util.PerConstants#UTF_8 `

File: platform/commons/src/main/java/com/peregrine/commons/servlets/AbstractBaseServlet.java (hunk 1 above)
Comment (reggie7) [reply platform/commons/src/main/java/com/peregrine/commons/servlets/AbstractBaseServlet.java]: I'd just change _decoding_ in the message do _encoding_. I don't know about the `logger` level - I'm pretty sure it's code that will virtually never happen.

**Review by reggie7 — APPROVED** [review]

File: admin-base/ui.apps/src/main/js/apiImpl.js [hunk 3]
```diff
@@ -661,7 +661,7 @@ class PerAdminImpl {
       let data = new FormData()
       data.append('fromTenant', fromName)
       data.append('toTenant', toName)
-      data.append('tenantTitle', tenantTitle)
+      data.append('tenantTitle', new Blob([tenantTitle], {type: 'application/json; charset=utf-8'}))
```
Comment (flx-sta) [inline admin-base/ui.apps/src/main/js/apiImpl.js]: There is an existing function `json(data)`

You wanna use that instead (DRY)

like that
```javascript
  data.append('tenantTitle', json(tenantTitle))
```

**Review by flx-sta — CHANGES_REQUESTED** [review]

File: admin-base/ui.apps/src/main/js/apiImpl.js (hunk 3 above)
Comment (reusr1) [reply admin-base/ui.apps/src/main/js/apiImpl.js]: was almost working with that function, added a new blob function to handle this usecase

**Review by flx-sta — APPROVED** [review]
image
image

**Pull request merged**